File: reactapp/src/models/Product.ts
export interface Category {
    id: number;
    title: string;
}

export interface Product {
    id: number;
    title: string;
    imageUrl: string;
    price: number;
    cart: number;
}

export interface ProductDetail {
    id: number;
    title: string;
    categoryId: number;
    category: string;
    description: string;
    favorite: boolean;
    types: ProductType[];
}

export interface ProductType {
    id: number;
    title: string;
    price: number;
    image: boolean;
    cart: number;
}

File: webapi/Models/ProductModels/Product.cs
namespace webapi.Models.ProductModels
{
    public class Product
    {
        public int Id { get; set; }
        public int CategoryId { get; set; }
        public string Category { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public IEnumerable<ProductType> Types { get; set; }
        public bool Favorite { get; set; }
    }

    public class ProductType
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public decimal Price { get; set; }
        public int Cart { get; set; }
        public bool Image { get; set; }
    }
}
